File: 2/python/part1.py
import sys

horizontal_position = 0
depth = 0

for line in sys.stdin:
    instruction, distance = line.rstrip().split(' ')

    if instruction == "forward":
        horizontal_position += int(distance)
    elif instruction == "down":
        depth += int(distance)
    elif instruction == "up":
        depth -= int(distance)

print(horizontal_position)
print(depth)
print(horizontal_position * depth)
    

File: 2/js/part1.js
const fs = require('fs');

const instructionToMotions = {
  'forward': d => ([d, 0]),
  'up': d => ([0, -d]),
  'down': d => ([0, d]),
};

fs.readFile(process.argv[2], 'utf-8', (err, data) => {
  console.log(
    data.split("\n")
      .map(s => s.split(' '))
      .map(([instruction, s]) => ([instruction, parseInt(s)]))
      .map(([instruction, d]) => instructionToMotions[instruction](d))
      .reduce(([h, d], [horizontal, depth]) => ([h + horizontal, d + depth]), [0, 0])
      .reduce((acc, n) => acc * n, 1)
  );
});
